test/integration/crud/maxTimeMS.test.ts

```diff
@@ -0,0 +1,266 @@
+import { expect } from 'chai';
+import { inspect } from 'util';
+
+import {
+  Collection,
+  CommandStartedEvent,
+  FindCursor,
+  MongoClient,
+  MongoCursorExhaustedError,
+  MongoServerError
+} from '../../../src';
+import { assert as test, setupDatabase } from '../shared';
+
+describe('MaxTimeMS', function () {
+  before(function () {
+    return setupDatabase(this.configuration);
+  });
+
+  it('should correctly respect the maxTimeMS property on count', function (done) {
```

Comment:
can we do async/await in these tests? moving them to ts already effectively makes it a brand new file to review and it's only a handful of tests so shouldn't add much scope to refactor but will make it easier to review the actual tests

Reply:
Done!